packages/docusaurus/src/commands/writeHeadingIds.ts

```diff
@@ -38,77 +38,60 @@ function addHeadingId(
 
   const headingText = line.slice(headingLevel).trimEnd();
   const headingHashes = line.slice(0, headingLevel);
-  const slug = slugger
-    .slug(unwrapMarkdownLinks(headingText).trim(), {maintainCase})
-    .replace(/^-+/, '')
```

Comment:
This also has the possibility of generating colliding slugs, so it's removed. We would probably fix it inside the `slugger.slug` function by first filtering out the illegal characters